components/kyma-environment-broker/internal/broker/instance_deprovision.go

```diff
@@ -64,8 +64,9 @@ func (b *DeprovisionEndpoint) Deprovision(ctx context.Context, instanceID string
 		logger.Errorf("cannot get existing operation from storage %s", errStorage)
 		return domain.DeprovisionServiceSpec{}, fmt.Errorf("cannot get existing operation from storage")
 
-		// there is an ongoing operation and it is not a temporary deprovision (suspension)
-	case existingOperation != nil && !existingOperation.Temporary && existingOperation.State != domain.Failed:
+		// there is an ongoing operation, and it is not a temporary deprovision (suspension)
+	case !(existingOperation == nil || existingOperation.Temporary || existingOperation.State == domain.Failed || existingOperation.State == domain.Succeeded):
```

Comment:
perhaps also check this filed `executedButNotCompleted` introduced by https://github.com/kyma-project/control-plane/pull/2412. KEB should probably only retrigger if the last deprovisioning that succeeded has any steps that need to be re-run. At that point, it might be a good idea from a readability perspective to encapsulate the condition in a function.

Reply:
This encapsulation crossed my mind. Good point.

Reply:
Conditioned extracted